### PR TITLE
Add providerConfig management

### PR DESCRIPTION
**Pull request by vshnbot**
## Summary

* Each claim/composite pair can specific their own providerConfig name
* It will be injected into each managed resource
* The providerConfigs need to be correctly provisioned beforehand
* Not setting the label will result in the current defaults being used
* Some slight refactoring for the Minio kindev configuration

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [ ] Link this PR to related issues.

Link: https://github.com/vshn/appcat/pull/237

### CODE REVIEW

File: tests/golden/apiserver/appcat/appcat/10_provider_sql.yaml [hunk 1]
```diff
@@ -45,18 +45,6 @@ metadata:
     name: crossplane-provider-provider-sql-system-custom
   name: crossplane:provider:provider-sql:system:custom
 rules:
-  - apiGroups:
```
Comment (Kidswiss) [inline tests/golden/apiserver/appcat/appcat/10_provider_sql.yaml:48]: This rule is invalid and never actually worked. Thus it's not necessary.

File: docs/modules/ROOT/pages/references/component-parameters.adoc [hunk 2]
```diff
@@ -67,34 +67,106 @@ reference:: xref:references/slo-parameters.adoc[See SLO parameters]
...
+required:: true
+
+The namespace of the secret.
+
+==== `data`
+type:: map
+default:: `{}`
+required:: true
+
+The contents of the `data` map of the secret.
```
Comment (zugao) [inline docs/modules/ROOT/pages/references/component-parameters.adoc:127]: I guess here you will have the details on how to connect to a different cluster? If yes can you elaborate more?

**Review by zugao — APPROVED** [review]
LGTM

File: docs/modules/ROOT/pages/references/component-parameters.adoc (hunk 2 above)
Comment (Kidswiss) [reply docs/modules/ROOT/pages/references/component-parameters.adoc:127]: This varies from provider to provider. So you'd have to consult the provider's docs.

**Pull request merged**